### ISSUE TITLE
Problem with handling big files

### ISSUE THREAD

**Issue by NicoxDJ** [blueimp/jQuery-File-Upload#3620]
Hello, i'm trying to use the system to upload big WAV files.
So i'm truing to upload 1.5GB WAV file, and in the beginning is going pretty well.
Its uploading fast (because its local uploading), but after the progress bar hit 100%, its just like... still trying to upload something more. Its continue to try like 3-4 minutes, and than returns error: "Empty file upload result". And i don't really know what is happening with this file. Is it still staying somewhere in TEMP?

I'm using:
jQuery File Upload 10.7.0
PHP 7.3
Safari

If you need other kind of information, just tell me what you need.

Best regards,
Nikolay Dimitrov

**Comment by blueimp** [comment]
When the progress bar hits 100%, the file upload itself has completed (for browsers with support for XHR progress events).
Afterwards, the browser waits for the request to complete, which depends on the processing time of the server.
A good UI improvement would be to indicate the differences in uploading and server-side processing.

In your case, assuming you're making use of the provided PHP demo script, it's likely the time it takes to move the file from PHPs temporary upload storage to its new location, using `move_uploaded_file`:
https://github.com/blueimp/jQuery-File-Upload/blob/master/server/php/UploadHandler.php#L1165
Ideally, you can configure your server in a way that both locations are on the same disk, which should make moving the file instantaneous.

You're likely hitting some timeout or another reason why the response doesn't return the actual result.

**Comment by NicoxDJ** [comment]
> Ideally, you can configure your server in a way that both locations are on the same disk, which should make moving the file instantaneous.
> 
> You're likely hitting some timeout or another reason why the response doesn't return the actual result.

Well, i'm using the same disk for storing, and its NVME SSD. Should be fast.
I think my timeout is 600 sec, i will double check the config.
And i will try to debug around move_uploaded_file() to see if something is happening there.

**Comment by NicoxDJ** [comment]
Nope, i can't debug it even properly. If i stop the script around the function "die();" it still continues to try. I try to debug in the post() function too, in the end $files was empty :/

**Comment by NicoxDJ** [comment]
It worked... somehow. But it's really really really slow. Like, the green progress bar fill for like 60-90 seconds, and than waiting ~15 minutes to move the 1.5GB file...  Something is wrong with the php side processing. :(

**Comment by blueimp** [comment]
Have you checked the following `php.ini` configs?
```
file_uploads = On
memory_limit = 2G
upload_max_filesize = 2G
post_max_size = 2G
max_execution_time = 1200
```

**Comment by NicoxDJ** [comment]
```
 file_uploads = On
 memory_limit = -1 (i have 16gb)
 upload_max_filesize = 5G
 post_max_size = 5G
 max_execution_time = 600
 ```
And as i said, for computer with NVMe SSD, and in the local network. The progress bar is filling from 0 to 100% for 60-90 sec, and than i want around 10-15min so 1.5GB file to be uploaded and refresh the file list.

**Comment by blueimp** [comment]
It's definitely possible that there's an issue with the sample PHP Upload handler.
Unfortunately I could not reproduce it so far, but I appreciate any investigation you or someone else can do into this.

**Comment by precision1980** [comment]
> 
> 
> ```
> file_uploads = On
> memory_limit = -1 (i have 16gb)
> upload_max_filesize = 5G
> post_max_size = 5G
> max_execution_time = 600
> ```
> 
> And as i said, for computer with NVMe SSD, and in the local network. The progress bar is filling from 0 to 100% for 60-90 sec, and than i wait around 10-15min so 1.5GB file to be uploaded and refresh the file list.

Hi NicoxDJ - what is your system config? I have been playing with this for a minute now and had some issues like you're describing at first. Once I changed my PHP memory to 2048m my time was a lot better and when I added 2 CPU's I have no problem even with files that are 2tb. This is a very sweet application by the way @blueimp . I started writing my own and then found yours and was like, well no need to re-invent the wheel! 

@blueimp - I also edited your sessions function to work with my DB sessions and uploading. to prevent unauthorized users from accessing the JSON output.
